packages/react-vapor/src/components/validation/components/ValidationMessage.tsx

```diff
@@ -0,0 +1,41 @@
+import * as React from 'react';
+import {connect} from 'react-redux';
+import {IReactVaporState} from '../../../ReactVapor';
+import {ValidationSelectors} from '../ValidationSelectors';
+
+export interface IValidationMessageProps {
+    id: string;
+}
+
+const mapStateToProps = (state: IReactVaporState, ownProps: IValidationMessageProps) => ({
+    errors: ValidationSelectors.getErrors(ownProps.id)(state),
+    warnings: ValidationSelectors.getWarnings(ownProps.id)(state),
+});
+
+export const ValidationMessageClasses = {
+    error: 'text-red',
+    warning: 'text-yellow',
+};
+
+export const ValidationMessageDisconnect: React.FunctionComponent<IValidationMessageProps &
+    ReturnType<typeof mapStateToProps>> = ({errors, warnings}) => {
+    return (
+        <>
+            {errors.length > 0
+                ? errors.map(({subId, value}) => (
+                      <span key={subId} className={ValidationMessageClasses.error}>
+                          {value}
+                      </span>
+                  ))
+                : warnings
+                ? warnings.map(({subId, value}) => (
+                      <span key={subId} className={ValidationMessageClasses.warning}>
+                          {value}
+                      </span>
+                  ))
+                : null}
```

Comment:
You really should avoid the nested ternaries it's hard to read and can be simplified with the classNames library

Reply:
I don't know, this looks weird to me:

```
const hasErrors = errors.length > 0;
const eitherErrorsOrWarnings = hasErrors ? errors : warnings;
return eitherErrorsOrWarnings.map((...) => <div className={classNames({
   "text-red": hasErrors
   "text-yellow": !hasErrors
])></div>);
```

It feels wrong to assume `errors` and `warnings` have the exact same signature (even though they do)

Any other idea to make it look good?

Reply:
I personally prefer the quote you did with `ClassNames`